### ISSUE TITLE
Document Elastic Agent Error messages and their description

### ISSUE THREAD

**Issue by nimarezainia** [elastic/observability-docs#782]
The Elastic agent could hit a number of errors during the course of operation which would move it to an "Unhealthy" state. The request here is to document what the errors may be and what they really mean so that the user has more context and can address the issue.

**Comment by nimarezainia** [comment]
Error in communication with fleet:
    - execute retry loop was stopped

Configuration processing errors:
    - failed to dispatch actions, error {error detail}
    - operator: received unexpected event
    - operator: failed to execute step {step name}, error: {error detail} // fails to download/start/install application
    - operation '%s' failed to verify
    - operation '%s' marked '%s.%s' corrupted

Installer specific errors:
    - tar contained invalid filename
    - TarInstaller: creating directory for file {file}: {error detail}
    - TarInstaller: creating file {file}: {error detail}
    - errors returned from tar/zip/filesystem

Downloader specific errors:
    - invalid upstream URI
    - generating package name failed: {error detail}
    - generating package path failed: {error detail}
    - fetching package failed: {error detail}
    - creating package failed: {error detail}
    - call to '{url}' returned unsuccessful status code: {status code}
    - package '{package}' not found: {error detail}

Capabilities filtering
    - input '%s' is not run due to capability restriction '%s'
    - output '%s' is left out due to capability restriction '%s'
    - upgrade is blocked due to capability restriction '%s'

Process configuration failures
    - application is not running
    - application stopping
    - yaml unmarshalling errors
    - failed to restart: {error detail}

Start related errors
    - exec: Stdin already set
    - exec: StdinPipe after process started
    - syscall error when creating pipe for stdin
    - failed to start '{error detail}'

**Comment by dedemorton** [comment]
Added the needs-input label because we need more input from the PM or product development teams before we can begin work on this issue. Specifically we need:

* An up-to-date list of the error messages that can cause agent to become unhealthy.
* A description of "what they really mean".
* Possibly what the user needs to do to resolve the error (if not obvious).

Keep in mind that this list will need to be maintained over time and it's probably better for the development team to "own" the updates because they'll know when things change.

IMO...error messages that need to be described in documentation are not well-written error messages. I think we should put more effort into improving the messages rather than trying to document messages that are confusing. For errors that are difficult to decipher because there are multiple root causes, we might want to add something to the troubleshooting section.

**Comment by dedemorton** [comment]
Moving this to the on hold/blocked column in the ingest-docs project because we have not gotten the input we need to write these docs.
